python/ambassador/fetch/ingress.py

```diff
@@ -127,8 +127,9 @@ def _process(self, obj: KubernetesObject) -> None:
             )
             return
 
-        # We don't want to deal with non-matching Ambassador IDs
-        if obj.ambassador_id != Config.ambassador_id:
+        # Ambassador ID, if specified, must match on either IngressClass or Ingress.
+        # The IngressClass, if we have one here, has already been checked for Ambassador ID in IngressClassProcessor._process
+        if not has_ingress_class and obj.ambassador_id != Config.ambassador_id:
```

Comment:
It seems this isn't enough to do what I intended, ie. avoid having to set the annotation on both ingress and ingressclass. I rarely have time to look at this though and last time I did I was a bit stuck, as in I don't know where to start.

I'll either need to figure out why this isn't enough or revert this part so that the remaining changes can move forward.

Reply:
I've changed the PR to only deal with this issue now, so I'll just need to find some time to check this again to see if i can get it working.